main.tf

```diff
@@ -14,16 +14,5 @@ locals {
   external_git_ref                 = join("", data.external.git.*.result.ref)
   git_ref                          = var.git_ref == "" ? local.external_git_ref : var.git_ref
   output_file                      = join("/", [var.module_path, var.filename])
-  url                              = join("", data.template_file.url.*.rendered)
-}
-
-data "template_file" "url" {
-  count    = module.this.enabled ? 1 : 0
-  template = replace(var.url, "$$", "$")
-
-  vars = {
-    filename    = var.filename
-    git_ref     = local.git_ref
-    module_name = var.module_name
-  }
+  url                              = replace(replace(replace(var.url, "$$${module_name}", var.module_name), "$$${git_ref}", var.git_ref), "$$${filename}", var.filename)
```

Comment:
@nitrocode any update on this?

We're running into template provider issues as well and would be willing to submit a PR with the requested changes if that would move this morning. 

cc @bwmetcalf

Reply:
@bwmetcalf @woz5999 just seeing ya'lls conversation it. Let's go with @nitrocode's suggestion -- That is cleaner and should be simple enough to make happen instead of delaying and implementing a sub-par solution. 

Put it up and I'll review 👍